lib/logstash/outputs/elasticsearch.rb

```diff
@@ -403,7 +403,7 @@ def register
       :upsert => @upsert,
       :doc_as_upsert => @doc_as_upsert
     }
-    common_options.merge! update_options if @action == 'update'
+    common_options.merge! update_options
```

Comment:
With some review : 
 - common_options[:upsert] is unused so it can be removed
 - we can totally remove this part if we consider that @doc_as_upsert could contains interpolation.(but needs more changes :) )